File: Beecrowd/1 - Beginner/2003 - 2176/2162 - Peaks and Valleys/2162_PeaksandValleys.py
landscapeMeasures = int(input())
height = list(map(int, input().split()))

count1 = count2 = 0

if height[0] < height[1]:
    for i in range(landscapeMeasures - 1):
        if i % 2 == 0:
            if height[i] < height[i + 1]:
                count1 += 1
        else:
            if height[i] > height[i + 1]:
                count2 += 1
elif height[0] > height[1]:
    for i in range(landscapeMeasures - 1):
        if i % 2 == 0:
            if height[i] > height[i + 1]:
                count1 += 1
        else:
            if height[i] < height[i + 1]:
                count2 += 1

if count1 + count2 + 1 == landscapeMeasures:
    print("1")
else:
    print("0")

File: Beecrowd/1 - Beginner/2003 - 2176/2162 - Peaks and Valleys/2162_PeaksandValleys.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int landscapeMeasures, count1 = 0, count2 = 0;
    cin >> landscapeMeasures;

    int height[landscapeMeasures];
    for (int i = 0; i < landscapeMeasures; i++) {
        cin >> height[i];
    }

    if (height[0] < height[1]) {
        for (int i = 0; i < landscapeMeasures - 1; i++) {
            if (i % 2 == 0) {
                if (height[i] < height[i + 1]) {
                    count1++;
                }
            } else {
                if (height[i] > height[i + 1]) {
                    count2++;
                }
            }
        }
    } else if (height[0] > height[1]) {
        for (int i = 0; i < landscapeMeasures - 1; i++) {
            if (i % 2 == 0) {
                if (height[i] > height[i + 1]) {
                    count1++;
                }
            } else {
                if (height[i] < height[i + 1]) {
                    count2++;
                }
            }
        }
    }

    if (count1 + count2 + 1 == landscapeMeasures) {
        cout << "1" << endl;
    } else cout << "0" << endl;

    return 0;
}